macro/src/lib.rs

```diff
@@ -86,7 +86,7 @@ fn gen_field(path: String, leaf: Field, out: &mut Vec<Item>) {
                         panic!("Option<Vec> is not supported");
                     }
 
-                    let field_name = leaf.ident.unwrap().to_string();
+                    let field_name = leaf.ident.map(|v| v.to_string()).unwrap_or(path);
```

Comment:
To match how we handle this in other cases, we should fall back to using the index of the field in the enum. That way, we don't conflate field names with non-terminal names.

Reply:
Thank you for your review! I've changed codes to fall back to field index!